### ISSUE TITLE
Optimal batch size

### ISSUE THREAD

**Issue by JunzheJosephZhu** [facebookresearch/dinov2#349]
Hi!
I'm trying to train dinov2 on a dataset with 14M images.
Two things I'm considering:
1. loading and freezing a pretrained dinov2 backbone, train the prototype heads for a few epochs, unfreeze the backbone, train the whole model. Apparently this requires vit-g since other archs are not trained with masked image modelling
2. just training a vit-L from scratch
In either of these cases, what's the recommended batch size? I assume for sinkhorn-knopp to work properly it needs to be quite big. The paper uses 2k-3k images per batch, but do you have a empirical minimum batch size that works reasonably well?
@TimDarcet @patricklabatut @pdollar @qasfb 
Would highly appreciate an answer!

**Comment by TimDarcet** [comment]
See https://github.com/facebookresearch/dinov2/issues/173#issuecomment-1878891181 in #173
